### ISSUE TITLE
DllNotFoundException during SavePayload execution

### ISSUE THREAD

**Issue by osharper** [natsuite/NatShare-API#100]
Hi, I just strated to use NatShare together with NatCorder in my app, love the latter so far but when I added NatShare plugin in order to save recorded video to Camera Roll, I encountered this exception:

```
DllNotFoundException: NatShare
NatSuite.Sharing.SavePayload+<>c__DisplayClass0_0.<.ctor>b__0 (NatSuite.Sharing.Internal.Bridge+CompletionHandler callback, System.IntPtr context) (at Assets/NatSuite/Plugins/Managed/Sharing/SavePayload.cs:22)
NatSuite.Sharing.Internal.NativePayload..ctor (System.Func`3[T1,T2,TResult] payloadCreator) (at Assets/NatSuite/Plugins/Managed/Sharing/Internal/NativePayload.cs:21)
NatSuite.Sharing.SavePayload..ctor (System.String album) (at Assets/NatSuite/Plugins/Managed/Sharing/SavePayload.cs:22)
```

I'm getting the following exception while launching my app in Unity Editor. I still haven't managed to set up debugging on iPad device (which is the primary target of the app) but something is failing there as well, I'm not sure whether it's the same exception or not. I'm using NatShare 1.2.3 in Unity 2020.1.9f1. I verified package import with Package manager and it says that there all files are already imported into the project.

Any ideas how to fix that? Thanks.

**Comment by olokobayusuf** [comment]
NatShare only works on mobile devices. Currently, there is no check to prevent NatShare from trying to load the native library and execute the calls on other platforms, but I'll add this check in the next update.

**Comment by osharper** [comment]
thanks for such quick response! really appreciate it. okay, so NatShare code cannot run in Unity Editor, right? okay, I can skip it for editor case. any ideas about iPad though?

**Comment by olokobayusuf** [comment]
> okay, so NatShare code cannot run in Unity Editor, right?

Yeah, it won't work in the Editor.

> any ideas about iPad though?

NatShare works on the iPad. If you have any issues let me know.

**Comment by olokobayusuf** [comment]
Fixed in #97
